### DISCUSSION TITLE
Weird physics behabiour with spheres

### DISCUSSION THREAD

**Discussion by vsirvent** [DanielChappuis/reactphysics3d#325] [Q&A]
Hi,

I'm using reactphysics3d in the game engine I'm developing and the physics works great most of the time, however I have a weird behaviour I show in the video link when a sphere is close to an edge, the terrain is generated as a concave mesh shape and the sphere with an sphere shape.

Any idea what can be happening? 

[Video showing the sphere issue](https://youtu.be/Ky3hOtIVa0Y)

Thank you for this great physics engine.

Vicente

**Comment by DanielChappuis** [comment]
Thanks for your feedback.

When you create the TriangleVertexArray for the ConcaveMeshShape, do you provide your own vertices normals or not? If you don't provide normals they are computed automatically by the library. I am wondering that maybe some normals are not correct at the vertices in the edges or your ConcaveMeshShape. That might cause this issue.

**Reply by vsirvent** [reply]
Hi Daniel,

Thank you for the reply, I'm passing the normals to the library. Those normals comes from the FBX file, do you recommend to let the library calculate automatically the normals?

**Reply by DanielChappuis** [reply]
The issue might come from the mesh or normals at that location. Maybe you can try using the automatic normals computation to see if it seems better or not.

**Reply by vsirvent** [reply]
Hi Daniel, I changed the balls locations and now I can't reproduce the issue anymore :( If I reproduce it again I'll try to let the engine to calculate its own normals.

Thank you very much for the support.
